cmd/frontman/main.go

```diff
@@ -338,12 +338,28 @@ sudo sysctl -w net.ipv4.ping_group_range="0   2147483647"`
 	interruptChan := make(chan struct{})
 	doneChan := make(chan struct{})
 
+	if fm.PidFile != "" && *oneRunOnlyModePtr && runtime.GOOS != "windows" {
+		err := ioutil.WriteFile(fm.PidFile, []byte(strconv.Itoa(os.Getpid())), 0664)
+		if err != nil {
+			log.Errorf("Failed to write pid file at: %s", fm.PidFile)
+		}
+	}
+
 	if *oneRunOnlyModePtr == true {
-		fm.Run(inputFilePtr, output, interruptChan, true)
+		input, err := fm.FetchInput(inputFilePtr)
+		if err != nil {
+			log.Error(err)
+			return
+		}
+
+		err = fm.RunOnce(input, output, interruptChan, false)
+		if err != nil {
+			log.Error(err)
```

Comment:
Why do we `return` in the error case before but not in this one?
If it's an indirect return done on line 359 please add a return into the error handling